smarts/core/tests/test_mutiple_traffic_data.py

```diff
@@ -117,14 +138,22 @@ def test_mutiple_traffic_data(create_scenario):
     use_first_history = True
     for scenario in scenarios:
         if use_first_history:
-            assert scenario.traffic_history is traffic_history_1
+            for key in scenario.traffic_history_service.traffic_history:
+                assert (
+                    scenario.traffic_history_service.traffic_history[key]
+                    == traffic_history_1[key]
+                )
         else:
-            assert scenario.traffic_history is traffic_history_2
+            for key in scenario.traffic_history_service.traffic_history:
+                assert (
+                    scenario.traffic_history_service.traffic_history[key]
+                    == traffic_history_2[key]
+                )
 
         traffic_history_provider.setup(scenario)
         assert (
-            traffic_history_provider._current_traffic_history
-            == scenario.traffic_history
+            traffic_history_provider._traffic_history_service
+            == scenario.traffic_history_service
         )
 
         use_first_history = not use_first_history
```

Comment:
Rather this weird switching logic I think it would be better to pair up the scenarios with their respective traffic like:

```python
for scenario, traffic_history in zip(scenarios, [traffic_history_1, traffic_history_2]):
```

Alternatively parameterise the fixtures.